server/test/unit/server/db/test_manage.py

```diff
@@ -141,10 +143,31 @@ def test_ensure_database_indexes(self, mock_model, mock_plugin_manager):
         the appropriate platform models
         """
         test_model = MagicMock()
-        mock_plugin_manager.return_value.unit_models.itervalues.return_value = [test_model]
+        mock_plugin_manager.return_value.unit_models.items.return_value = [('test-unit',
+                                                                            test_model)]
         manage.ensure_database_indexes()
         test_model.ensure_indexes.assert_called_once_with()
 
+    @patch.object(manage, 'PluginManager')
+    @patch.object(manage, 'model')
+    def test_ensure_database_indexes_throws_exception(self, mock_model, mock_plugin_manager):
+        """
+        Make sure that the ensure_indexes method is called for all
+        the appropriate platform models
+        """
+        test_model = MagicMock()
+        test_model.unit_key_fields = ('1', '2', '3')
+        unit_key_index = {'fields': test_model.unit_key_fields, 'unique': True}
+        test_model._meta.__getitem__.side_effect = [[unit_key_index]]
+        mock_plugin_manager.return_value.unit_models.items.return_value = [('test-unit',
+                                                                            test_model)]
+        try:
```

Comment:
It's usually simpler and more readable to use ``self.assertRaises`` as a context manager (which does require the use of ``pulp.common.compat.unittest``) Off the top of my head, it's something like

```
with self.assertRaises(PulpCodedException) as context:
  manage.ensure_database_indexes()
self.assertEqual(context.exception.error_code, error_codes.PLP0044)